src/components/Greeting/index.js

```diff
@@ -1,31 +1,55 @@
 // @flow
 import React from 'react';
+import { setIsLoggedIn } from 'pages/SignIn/actions';
+import type { Action } from 'pages/SignIn/actions';
+import { connect } from 'react-redux';
+import firebase from 'config';
+
+const mapDispatchToProps = {
+  setIsLoggedIn,
+};
 
 type Props = {|
   userInfo: UserInfo,
+  setIsLoggedIn: (isLoggedIn: boolean) => Action,
+  setUserInfo: (userInfo: ?UserInfo) => Action,
 |};
 
 const styles = {
   box: {
     display: 'flex',
     justifyContent: 'space-between',
   },
+  logout: {
+    backgroundColor: '#c3e6bc',
+    border: '2px solid darkgreen',
+    borderRadius: '2px',
+    color: 'darkgreen',
+    fontSize: '20px',
+    fontWeight: 'bold',
+  },
   text: {
     color: 'darkgreen',
     fontSize: '25px',
     fontStyle: 'italic',
   },
 };
 
-const Greeting = (props: Props) => {
-  const { userInfo: { email } } = props;
+const LoggedInHeader = (props: Props) => {
+  const { userInfo } = props;
+  const email = userInfo ? userInfo.email : null;
   const logOutMessage = 'Log Out';
+  const logOut = async () => {
+    await firebase.auth().signOut();
```

Comment:
move this (and other firebase functions) to `src/services/firebase` and import it from there